core/src/main/java/org/infinispan/commands/statetransfer/AbstractStateTransferCommand.java

```diff
@@ -0,0 +1,78 @@
+package org.infinispan.commands.statetransfer;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+import org.infinispan.commands.TopologyAffectedCommand;
+import org.infinispan.commands.remote.BaseRpcCommand;
+import org.infinispan.commons.util.IntSet;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.util.ByteString;
+
+/**
+ * Base class for commands related to state transfer.
+ *
+ * @author Ryan Emerson
+ * @since 11.0
+ */
+abstract class AbstractStateTransferCommand extends BaseRpcCommand implements TopologyAffectedCommand {
+
+   private byte commandId;
+   protected int topologyId;
+   protected IntSet segments;
+
+   AbstractStateTransferCommand(byte commandId, ByteString cacheName) {
+      super(cacheName);
+      this.commandId = commandId;
+   }
+
+   AbstractStateTransferCommand(byte commandId, ByteString cacheName, Address origin, int topologyId, IntSet segments) {
+      this(commandId, cacheName);
+      this.origin = origin;
```

Comment:
`origin` is always `rpcManager.getAddress()`. 
It is not necessary to set it in the constructor neither sending it through the wire.